scripts/west_commands/export.py

```diff
@@ -39,38 +42,52 @@ def do_add_parser(self, parser_adder):
-
-        # Let's clean up, as Zephyr has now been exported, and we no longer
-        # need the generated files.
-        cmake_args = ['--build', f'{zephyr_config_package_path}',
-                      '--target', 'pristine']
-        run_cmake(cmake_args, capture_output=True)
-
-        # Let's ignore the normal CMake printing and instead only print
-        # the important information.
-        msg = [line for line in lines if not line.startswith('-- ')]
-        print('\n'.join(msg))
-
-        zephyr_unittest_config_package_path = PurePath(__file__).parents[2] \
-            / 'share' / 'zephyrunittest-package' / 'cmake'
-
-        cmake_args = ['-S', f'{zephyr_unittest_config_package_path}',
-                      '-B', f'{zephyr_unittest_config_package_path}']
-        lines = run_cmake(cmake_args, capture_output=True)
-
-        # Let's clean up, as Zephyr has now been exported, and we no longer
-        # need the generated files.
-        cmake_args = ['--build', f'{zephyr_unittest_config_package_path}',
-                      '--target', 'pristine']
-        run_cmake(cmake_args, capture_output=True)
-
-        # Let's ignore the normal CMake printing and instead only print
-        # the important information.
+        # The 'share' subdirectory of the top level zephyr repository.
+        share = Path(__file__).parents[2] / 'share'
+
+        if which('ninja') is not None:
+            generator = 'Ninja'
+        elif which('make') is not None:
+            generator = 'Unix Makefiles'
+        else:
+            log.die('neither ninja nor make is installed')
+
+        run_cmake_and_clean_up(str(share / 'zephyr-package' / 'cmake'),
+                               generator)
+        run_cmake_and_clean_up(str(share / 'zephyrunittest-package' / 'cmake'),
+                               generator)
+
+def run_cmake_and_clean_up(path, generator):
+    # Run a package installation script, cleaning up afterwards.
+    #
+    # Filtering out lines that start with -- ignores the normal
+    # CMake status messages and instead only prints the important
+    # information.
+
+    try:
+        lines = run_cmake(['-S', path, '-B', path, f'-G{generator}'],
+                          capture_output=True)
+    finally:
         msg = [line for line in lines if not line.startswith('-- ')]
-        print('\n'.join(msg))
+        log.inf('\n'.join(msg))
+        clean_up(path)
+
+def clean_up(path):
+    try:
+        run_cmake(['--build', path, '--target', 'pristine'],
+                  capture_output=True)
+    except CalledProcessError:
+        # Do our best to clean up even though CMake failed.
+        log.wrn(f'Failed to make {path} pristine; '
+                'removing known generated files...')
+        for subpath in ['CMakeCache.txt', 'CMakeFiles', 'build.ninja',
+                        'cmake_install.cmake', 'rules.ninja']:
+            remove_if_exists(Path(path) / subpath)
```

Comment:
unresolved comment here: https://github.com/zephyrproject-rtos/zephyr/pull/23977#discussion_r402955483

Reply:
when not using the build command, then there is only a chance that CMake itself fails, and in case that happens, I assume we are in deeper troubles, but let's keep it as a precaution.